<launch>
  <arg name="voc_file" default="$(find ORB_SLAM2)/Vocabulary/ORBvoc.bin"/>
  <arg name="settings_file" default="$(find ORB_SLAM2)/config/kinect.yaml"/>
  <arg name="rgb_topic" default="/camera/rgb/image_color"/>
  <arg name="depth_topic" default="/camera/depth/image"/>

  <node name="RGBDpub" pkg="ORB_SLAM2"  type="RGBDpub" 
        args = "$(arg voc_file) $(arg settings_file)" output="screen">

      <remap from="/camera/rgb/image_raw" to="$(arg rgb_topic)" />
      <remap from="/camera/depth_registered/image_raw" to="$(arg depth_topic)" />
       <param name="publish_pointcloud" type="bool" value="true" />
       <param name="publish_pose" type="bool" value="true" />
       <param name="localize_only" type="bool" value="false" />
       <param name="reset_map" type="bool" value="false" />

       <!-- static parameters -->
       <param name="load_map" type="bool" value="false" />
       <param name="map_file" type="string" value="map.bin" />
       <param name="settings_file" type="string" value="$(arg settings_file)" />
       <param name="voc_file" type="string" value="$(arg voc_file)" />

       <param name="pointcloud_frame_id" type="string" value="map" />
       <param name="camera_frame_id" type="string" value="camera_link" />
       <param name="min_num_kf_in_map" type="int" value="5" />

  </node>

</launch>